dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java

```diff
@@ -868,8 +872,20 @@ public void setTopicManual(String topicManual) {
         this.topicManual = topicManual;
     }
 
+    public String getTopicAI() {
+        return topicAI;
+    }
+
+    public void setTopicAI(String topicAI) {
+        this.topicAI = topicAI;
+    }
+
     public String getTopic() {
-        return this.topicSelection == TopicSelection.AUTOMATIC ? this.getTopicAutomatic() : this.getTopicManual();
+        return switch (this.topicSelection) {
```

Comment:
Just thinking, would we ever want to surface both? For example, as a workflow developer, I might try and craft a description manually, then let the AI generate one, and want to see what both look like, picking the better one. I know we're not generating AI descriptions dynamically, so this is future looking (and could be done in the future), but just raising it as a possibility.

Reply:
We do surface all three topic types via their individual fields, `topicManual`, `topicAutomatic`, and `topicAI`. This `getTopic()` function is used by the ElasticListener to get the topic based on the selection when creating an elastic entry:
https://github.com/dockstore/dockstore/blob/4c12b34709b9a94e74202a75629b7712efbe9c29/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java#L447